pkg/fake/ec2api.go

```diff
@@ -68,6 +68,9 @@ type EC2Behavior struct {
 	LaunchTemplates                     sync.Map
 	InsufficientCapacityPools           atomic.Slice[CapacityPool]
 	NextError                           AtomicError
+
+	// Tracks the capacity reservations associated with launch templates, if applicable
+	launchTemplateCapacityReservationIndex sync.Map
```

Comment:
```suggestion
	launchTemplateCapacityReservationIndex sync.Map // launch template name -> capacity reservation id
```

So when you say index, you mean like a DB index for quick lookup? We add a comment like this in `cluster.go` in upstream, this at least helps me quickly reason about the types and what the key and values are

Reply:
Actually, rather than storing a separate index, does it make sense to just re-leverage the current `LaunchTemplates` and just start storing the CR data inside of there -- if we had any other launch template data that we needed to add in the future, it would be pretty easily expandable, since we would always just be working with an LT

Reply:
I thought it was a little cleaner to store it separately since most launchtemplates will not have an associated capacity reservation, but I can go either way.

Reply:
I was more thinking that it's a bit of an implementation detail that we aren't actually storing the launch template data today -- we just as well could be and then you wouldn't need this map at all -- I'd opt to drop the map and we can expand the data that we store in our launch template data store over time as it's needed

Reply:
I'm not in love with it but we discussed offline and agreed to take this in a follow-on